File: ios/Types/PlacePickerOptions.swift
//
//  Options.swift
//  react-native-place-picker
//
//  Created by b0iq on 10/11/2022.
//

import ExpoModulesCore

struct PlacePickerOptions: Record {
    @Field
    var presentationStyle: PlacePickerPresentationStyle = .fullscreen
    @Field
    var title: String = "Choose Place"
    @Field
    var searchPlaceholder: String = "Search..."
    @Field
    var color: String = "FF0000"
    @Field
    var contrastColor: String = "FFFFFF"
    @Field
    var locale: String = "en-US"
    @Field
    var initialCoordinates: PlacePickerCoordinate = PlacePickerCoordinate(latitude: .init(wrappedValue: 25.2048), longitude: .init(wrappedValue: 55.2708))
    @Field
    var enableGeocoding: Bool = true
    @Field
    var enableSearch: Bool = true
    @Field
    var enableUserLocation: Bool = true
    @Field
    var enableLargeTitle: Bool = true
    @Field
    var rejectOnCancel: Bool = true
    @Field
    var enableRangeSelection: Bool = false
    @Field
    var initialRadius: Double = 1000
    @Field
    var minRadius: Double = 100
    @Field
    var maxRadius: Double = 10000
    @Field
    var radiusColor: String = ""
    @Field
    var radiusStrokeColor: String = ""
    @Field
    var radiusStrokeWidth: Double = 2
}

File: android/src/main/java/expo/modules/placepicker/Types/PlacePickerOptions.kt
package expo.modules.placepicker

import expo.modules.kotlin.records.Field
import expo.modules.kotlin.records.Record

class PlacePickerOptions : Record {
    @Field
    val presentationStyle: PlacePickerPresentationStyle = PlacePickerPresentationStyle.fullscreen

    @Field
    val title: String = "Choose Place"

    @Field
    val searchPlaceholder: String = "Search..."

    @Field
    val color: String = "#FF0000"

    @Field
    val contrastColor: String = "#FFFFFF"

    @Field
    val locale: String = "en-US"

    @Field
    val initialCoordinates: PlacePickerCoordinate? = null

    @Field
    val enableGeocoding: Boolean = true

    @Field
    val enableSearch: Boolean = true

    @Field
    val enableUserLocation: Boolean = true

    @Field
    val enableLargeTitle: Boolean = true

    @Field
    val rejectOnCancel: Boolean = true

    @Field
    val enableRangeSelection: Boolean = false

    @Field
    val initialRadius: Double = 1000.0

    @Field
    val minRadius: Double = 100.0

    @Field
    val maxRadius: Double = 10000.0

    @Field
    val radiusColor: String = ""

    @Field
    val radiusStrokeColor: String = ""

    @Field
    val radiusStrokeWidth: Double = 2.0
}
